src/commands/github/foldingProvider/foldingRangesMap.ts

```diff
@@ -0,0 +1,17 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { FoldingRange, Uri } from "vscode";
+
+const foldingRangeMap: { [key: string]: FoldingRange[] } = {}
+
+export function getFoldingRanges(uri: Uri): FoldingRange[] {
+    return foldingRangeMap[uri.fsPath];
```

Comment:
> Do you think exposing the uri for ReadOnlyContent would be okay?

I'm okay with it if you want a quick solution. It's not ideal because it's exposing an internal detail of how `openReadonlyContent` works. For example, we currently stick the unique `idHash` string in the path itself, but `idHash` could've easily gone in the query instead. If we ever switched it, your `fsPath` referenced here would no longer be unique.

I'm starting to feel like the ideal solution might be to create your own version of `ReadOnlyContentProvider`. I've changed my mind for a few reasons
1. It's not as much code as I originally remembered
2. The GitHub logic seems unique enough that it might be worth it. Specifically, you can be inspired by `ReadOnlyContentProvider`, but I wouldn't wanting you copying it over as-is. Ideally you create a new `TextDocumentContentProvider ` specifically for GitHub logs 
3. `openReadOnlyContent` is actually used in quite a few extensions, so I like keeping it simple and clean